src/lib.jl

```diff
@@ -244,6 +244,38 @@ function sinkhorn2(μ, ν, C, ε; kwargs...)
     return pot.sinkhorn2(μ, ν, PyCall.PyReverseDims(permutedims(C)), ε; kwargs...)
 end
 
+"""
+    empirical_sinkhorn_divergence(X_s, X_t, reg; kwargs...)
+
+Compute the Sinkhorn Divergence from empirical data, where
+`X_s` and `X_t` are arrays representing samples in the source domain and target domain, respectively,
+and  `reg` is the regularization term. `a` and `b` are optinal sample weights in the source
+and target domain, respectively.
+
+This function is a wrapper of the function
+[`ot.bregman.empirical_sinkhorn_divergence`](https://pythonot.github.io/all.html#ot.emd) in the Python Optimal Transport
+package. Keyword arguments are listed in the documentation of the Python function.
+
+```jldoctest empirical_sinkhorn_divergence
+julia> X_s = [1.0, 0.0, 3.0];
+
+julia> X_t = [2.0, 3.0, 4.0];
+
+julia> round(empirical_sinkhorn_divergence(X_s,X_t,1); sigdigits=3)
+3.03
+```
+
+See also: [`sinkhorn2`](@ref)
+"""
+function empirical_sinkhorn_divergence(X_s, X_t, reg; kwargs...)
+    return pot.bregman.empirical_sinkhorn_divergence(
+        typeof(X_s) <: Vector ? reshape(X_s, :, 1) : X_s,
+        typeof(X_t) <: Vector ? reshape(X_t, :, 1) : X_t,
```

Comment:
This is incorrect for more general vector types. Does it not work automatically or with PyReverseDims?

Reply:
It does not work automatically, it requires vectors to be `n x 1`  matrices. I never used the PyReverseDims.

Reply:
For what situation this doesn't work?

Reply:
If matrices are needed you can try
```suggestion
        reshape(xsource, Val(2)),
        reshape(xtarget, Val(2)),
```
This reshapes vectors (not only of type `Vector`) to matrices and keeps matrices as matrices.

Reply:
> For what situation this doesn't work?

If you use a vector that is not of type `Vector`.